src/odemis/driver/tescan.py

```diff
@@ -1127,12 +1161,22 @@ def _doMoveAbs(self, pos):
             logging.debug("Stage move to %s completed", pos)
             self._updatePosition()
 
-    def _doMoveRel(self, shift):
-        # Compute the position in absolute coordinates
+    def _doMoveRel(self, shift: dict):
+        """
+        Request the position in absolute coordinates and create new positions after applying shift values
+        :param shift (dict[str, float]): shift of linear axes in m or rotational axes in radians
+        """
         self._updatePosition()
         pos = self._position.copy()
+
+        # add the requested change to the current position
         for axis, change in shift.items():
             pos[axis] += change
+            if axis in ("rx", "rz"):
+                rng = self.axes[axis].range
+                if abs(rng[1] - rng[0]) > 2 * math.pi:
+                    # for full rotational axes this check maps values outside the range to between 0 and 2pi
+                    pos[axis] = (pos[axis] - abs(rng[0])) % (2 * math.pi)
```

Comment:
This code does not look correct. Unless rng[0] == 0 (which is probably very common), it will not adjust back the value to fit the range. I think this should work (same code as in xt_client):
```suggestion
                    pos[axis] = (pos[axis] - rng[0]) % (2 * math.pi) + rng[0]
```

Reply:
I understand. That code indeed somehow assumes that the total is range is 2π. However, with the hardware, you end up with 4π . I guess we could adjust it to use the actual range, using something like that: 
`(pos[axis] - rng[0]) % abs(rng[1] - rng[0]) + rng[0]`.
However, this might cause other weird behaviour if the hardware reports incorrect range which is not exactly a multiple of 2π. 

I'm not sure why they support such range in the first place. As they don't support relative moves, I could imagine that this is to allow indicating the direction of the rotation. For example, if the stage is at 0°, and you ask to move to -20°, it goes counter-clockwise, while if you ask to go to 340°, it would go to the same place but clockwise. At the end of the move it would always indicate 340°. However, to fully support this feature you'd need a range from -360° to 720°, as to support also going from 340° to 10° clockwise (this is what we actually do on some axes in Odemis). 
-360° to 360° seems too small to be useful to me. 

So maybe it does nothing special. In that case, a simpler way to handle it in Odemis is in the init, to check explicitly for the range -360° -> 360°, and replace it with just 0->2π. (Note that it's almost the same that we do in the xt_client, because there the SEM reports a range of 0->0 for axes that support full rotation).
If you do the init this way, then here you can more safely write: `(pos[axis] - rng[0]) % (2 * math.pi) + rng[0]`

Reply:
Allright thanks for sparring a bit over this matter, I chose the second option of your suggestion. Specifically because Tescan Essence reports values of the rz axis only in the 0 -> 360° range as well so I think we will be safe with this choice.